addons/mail/static/src/core/common/thread_model.js

```diff
@@ -1131,7 +1135,14 @@ export class Thread extends Record {
     }
 
     updateSeen(lastSeen = this.newestPersistentOfAllMessage) {
-        const lastReadIndex = this.messages.findIndex((message) => message.eq(lastSeen));
+        if (this.selfMember) {
+            this.selfMember.seen_message_id = lastSeen;
+        }
+        this.updateUnreadAndNeedactionCounters(lastSeen);
+    }
+
+    updateUnreadAndNeedactionCounters(messageFrom) {
```

Comment:
The logic in this function is completely flawed, I think it would be best if "updateSeen" (which should make an RPC) returned the proper new counters (on the bus), which would also fix multi-tab.

Updating seen value or counters locally makes no sense at all.

Reply:
When receiving a message from another member we still need to compute the counter, but it's the only flow.

When using mark as read for self or when self posting a message, the server should give the value.

Reply:
The race condition is normally fixed by bus_id, whenever we set the full value we update it, and we can check when we receive the message for its bus id too.